src/dsp/platform.c

```diff
@@ -1906,6 +1906,108 @@ int decode_numeric_sensor_pdr_data(
 	return PLDM_SUCCESS;
 }
 
+LIBPLDM_ABI_STABLE
+int decode_redfish_resource_pdr_data(const void *pdr_data,
+				     size_t pdr_data_length,
+				     struct pldm_redfish_resource_pdr *pdr_value)
+{
+	PLDM_MSGBUF_DEFINE_P(buf);
+	int rc;
+
+	if (!pdr_data || !pdr_data_length || !pdr_value) {
+		return PLDM_ERROR_INVALID_DATA;
+	}
+
+	rc = pldm_msgbuf_init_errno(buf,
+				    PLDM_PDR_REDFISH_RESOURCE_PDR_MIN_LENGTH,
+				    pdr_data, pdr_data_length);
+	if (rc) {
+		return pldm_xlate_errno(rc);
+	}
+
+	rc = pldm_msgbuf_extract_value_pdr_hdr(
+		buf, &pdr_value->hdr, PLDM_PDR_REDFISH_RESOURCE_PDR_MIN_LENGTH,
+		pdr_data_length);
+	if (rc) {
+		return pldm_xlate_errno(pldm_msgbuf_discard(buf, rc));
+	}
+
+	pldm_msgbuf_extract(buf, pdr_value->resource_id);
+	pldm_msgbuf_extract(buf, pdr_value->resource_flags.byte);
+	pldm_msgbuf_extract(buf, pdr_value->cont_resrc_id);
+
+	pldm_msgbuf_extract_uint16_to_size(buf,
+					   pdr_value->prop_cont_resrc.length);
+	pldm_msgbuf_span_required(buf, pdr_value->prop_cont_resrc.length,
+				  (void **)&pdr_value->prop_cont_resrc.name);
+
+	pldm_msgbuf_extract_uint16_to_size(buf, pdr_value->sub_uri.length);
```

Comment:
Length in bytes of the SubURI path fragment (including the null terminator) for the primary resource. please make sure this handle properly while extracting the suburi data.